lib/servers/Mulitplexer/Multiplexerserver.py

```diff
@@ -23,298 +16,481 @@
-        return self.error
-
-    def getIntegrator(self):
-        return self.Integrator
-
-    def getDerivator(self):
-        return self.Derivator
 
 class MultiplexerServer(LabradServer):
     """
-    Multiplexer Server for Wavelength Meter
+    Multiplexer server for WS-U wavelength meter.
+
+    A DLL is required to run this server.  See initServer for the path location
+    for the library.
     """
     name = 'Multiplexerserver'
 
-    #Set up signals to be sent to listeners
-    measuredchanged = Signal(CHANSIGNAL, 'signal: selected channels changed', '(ib)')
+    # Set up signals to be sent to listeners
+    channel_text = 'signal: selected channels changed'
+    measuredchanged = Signal(CHANSIGNAL, channel_text, '(ib)')
     freqchanged = Signal(FREQSIGNAL, 'signal: frequency changed', '(iv)')
     updateexp = Signal(UPDATEEXP, 'signal: update exp', '(ii)')
     lockchanged = Signal(LOCKSIGNAL, 'signal: lock changed', 'b')
     outputchanged = Signal(OUTPUTCHANGED, 'signal: output changed', 'b')
-    pidvoltagechanged = Signal(PIDVOLTAGE, 'signal: pidvoltage changed', '(iv)')
-    
+    voltage_text = 'signal: pidvoltage changed'
+    pidvoltagechanged = Signal(PIDVOLTAGE, voltage_text, '(iv)')
+    channellock = Signal(CHANNELLOCK, 'signal: channel lock changed', '(wwb)')
+    ampchanged = Signal(AMPCHANGED, 'signal: amplitude changed', '(wv)')
+
     def initServer(self):
 
-        #load wavemeter dll file for use of API functions self.d and self.l are dummy c_types for unused wavemeter functions
-        
-        self.d = c_double(0)
-        self.l = c_long(0)    
-        self.b = c_bool(0)    
-        self.wmdll = windll.LoadLibrary("C:\Windows\System32\wlmData.dll")
-        
-        #allocates c_types for dll functions
-        self.wmdll.GetActiveChannel.restype        = c_long 
-        self.wmdll.GetAmplitudeNum.restype         = c_long 
-        self.wmdll.GetDeviationMode.restype        = c_bool      
-        self.wmdll.GetDeviationSignalNum.restype   = c_double
-        self.wmdll.GetExposureNum.restype          = c_long
-        self.wmdll.GetFrequencyNum.restype         = c_double  
-        self.wmdll.GetPIDCourseNum.restype         = c_long
-        self.wmdll.GetSwitcherMode.restype         = c_long
-        self.wmdll.GetSwitcherSignalStates.restype = c_long
-
-        self.wmdll.SetDeviationMode.restype        = c_long
-        self.wmdll.SetDeviationSignalNum.restype   = c_double
-        self.wmdll.SetExposureNum.restype          = c_long 
-        self.wmdll.SetPIDCourseNum.restype         = c_long              
-        self.wmdll.SetSwitcherSignalStates.restype = c_long        
-        self.wmdll.SetSwitcherMode.restype         = c_long
-        self.wmdll.SetDeviationSignal.restype      = c_long
-        
+        # load wavemeter dll file for use of API functions self.d and self.l
+        # are dummy c_types for unused wavemeter functions
+
+        dll_path = "C:\Windows\System32\wlmData.dll"
```

Comment:
will this be a operating system problem? I guess as long as the wavemeter server is run on windows machines it should be fine.